DevOps course refactor (#1737)
rolling-scopes-school/tasks: devops/modules/2_cluster-configuration/task_3.md

# Task: K8s Cluster Configuration and Creation
![task_3 schema](../../visual_assets/task_3.png)
## Objective

In this task, you will configure and deploy a Kubernetes (K8s) cluster on AWS using or k3s. You will also verify the cluster by running a simple workload.

## Steps

1. **Choose Deployment Method**

   - Get familiar with [k3s](https://k3s.io/).

>    - Use AWS EC2 instances from the Free Tier to avoid additional expenses.

Reviewer: ```suggestion
   - Use AWS EC2 instances from the [AWS Free Tier](https://aws.amazon.com/free/) to avoid additional expenses.
```

3. **Deploy the Cluster**

>    - Deploy the K8s cluster using k3s.
>    - Ensure the cluster is accessible from your bastion host 
>    - Exra step to make it accessable from your local computer.

Reviewer: Do we actually want to deploy the cluster in private subnets?

```suggestion
   - Deploy [Bastion host](https://www.geeksforgeeks.org/what-is-aws-bastion-host/)
   - Deploy the K8s cluster using k3s.
   - Ensure the cluster is accessible from your bastion host 
   - **Additional task** make it accessible from your local computer.
```

   - Ensure the workload runs successfully on the cluster.

6. **Additional Tasks**
   - Document the cluster setup and deployment process in a README file.

## Submission

> - Create task_3 branch in your repository

Reviewer: ```suggestion
- Create `task_3` branch from `main` in your repository.
```

   - Terraform code is created or extended to manage AWS resources required for the cluster creation.
   - The code includes the creation of a bastion host.

2. **Cluster Verification (50 points)**

>    - The cluster is verified by running the `kubectl get nodes` command from the local computer.

Reviewer: local computer or bastion?

Author: Moved "...from local computer" to the additional tasks.

   - A screenshot of the `kubectl get nodes` command output is provided.

4. **Workload Deployment (30 points)**

   - A simple workload is deployed on the cluster using `kubectl apply -f https://k8s.io/examples/pods/simple-pod.yaml`.

>    - A screenshot of the `kubectl get all` command output is provided. (pod named "nginx" should be presented)

Reviewer: Probably, it's worth to move this to the `submission` part, and here write something like "check that pod named nginx is running in the cluster"?
